### PR TITLE
Use JSON instead of YAML

### PR DESCRIPTION
**Pull request by kaczmarj**
Addresses #9

### CODE REVIEW

**Comment by kaczmarj** [comment]
Build raises `APIError: 400 Client Error: Bad Request ("client is newer than server (client API version: 1.26, server API version: 1.24)")`. This is fixed by ab328ab70592c6e7f7d95445f8b1ce1612a58a3a in PR #10 .

Pytest passes on my local machine when running `python -m pytest /Users/jakubkaczmarzyk/Projects/nipype-regtests/src/`.

**Comment by mgxd** [comment]
rebuilding - will merge once passing

File: samples/ubuntu-16.04_py-3.5.1/Dockerfile [hunk 1]
```diff
@@ -7,15 +7,15 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
...
     LC_ALL=C.UTF-8
-COPY conda-env.yml /home/conda-env.yml
+COPY conda-env.json /home/conda-env.json
 RUN $CONDAPATH/conda update -y conda && \
-    $CONDAPATH/conda env create -f /home/conda-env.yml -n new_env && \
+    $CONDAPATH/conda env create -f /home/conda-env.json -n new_env && \
     cd $CONDAPATH/../ && rm -rf conda-meta include share ssl && \
     $CONDAPATH/conda clean -y -a
-ENV PATH=/usr/local/miniconda/envs/new_env/bin:$PATH
+ENV PATH=/usr/local/miniconda/envs/new_env/bin:$PATH
```
Comment (mgxd) [inline samples/ubuntu-16.04_py-3.5.1/Dockerfile:21]: new line

File: samples/ubuntu-16.04_py-3.5.1/conda-env.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "dependencies": [
+        "python=3.5.1",
+        "numpy",
+        {
+            "pip": [
+                "pandas"
+            ]
+        }
+    ]
+}
```
Comment (mgxd) [inline samples/ubuntu-16.04_py-3.5.1/conda-env.json:11]: again

File: src/utils.py [hunk 3]
```diff
@@ -46,19 +42,14 @@ def set_log_level(level):
...
-def save_yaml(obj, filepath):
-    """Save Python dictionary as YAML file."""
-    # Add overwrite option.
-    # if op.exists(filepath):
-    #     raise Exception("File already exists: {}".format(filepath))
-    with open(filepath, 'w') as stream:
-        yaml.dump(obj, stream, Dumper=yaml.RoundTripDumper, block_seq_indent=2)
+def save_json(obj, filepath):
+    with open(filepath, 'w') as fp:
+        json.dump(obj, fp, indent=4)
```
Comment (mgxd) [inline src/utils.py]: more flexible to define indent in arguments with a default of 4

**Review by mgxd — CHANGES_REQUESTED** [review]

File: samples/ubuntu-16.04_py-3.5.1/conda-env.json (hunk 2 above)
Comment (kaczmarj) [reply samples/ubuntu-16.04_py-3.5.1/conda-env.json:11]: This JSON was created with `src.utils.save_json()`. I updated that function to add a newline at the end of the file.

File: samples/ubuntu-16.04_py-3.5.1/Dockerfile (hunk 1 above)
Comment (kaczmarj) [reply samples/ubuntu-16.04_py-3.5.1/Dockerfile:21]: I'll keep this in mind when I update the class that creates the Dockerfile.

**Review by mgxd — APPROVED** [review]

**Pull request merged**